=== modified file 'openerp/addons/base/rng/view.rng' (properties changed: -x to +x)
--- openerp/addons/base/rng/view.rng	2012-02-13 10:17:58 +0000
+++ openerp/addons/base/rng/view.rng	2013-09-17 13:24:14 +0000
@@ -547,6 +547,7 @@
             <rng:optional><rng:attribute name="filters"/></rng:optional>
             <rng:optional><rng:attribute name="statusbar_visible"/></rng:optional>
             <rng:optional><rng:attribute name="statusbar_colors"/></rng:optional>
+            <rng:optional><rng:attribute name="mask"/></rng:optional>
             <!-- Widget *static* options defined as an arbitrary JSON dict, with
                  widget-dependent parameters. To be ignored if widget/client does
                  not support them. -->

